adapters/smaato/smaato.go

```diff
@@ -179,15 +180,27 @@ func (a *SmaatoAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRe
 			bid := sb.Bid[i]
 
 			var markupError error
-			bid.AdM, markupError = renderAdMarkup(getAdMarkupType(response, bid.AdM), bid.AdM)
+			markupType := getAdMarkupType(response, bid.AdM)
+			bid.AdM, markupError = renderAdMarkup(markupType, bid.AdM)
 			if markupError != nil {
-				fmt.Println(markupError)
+				continue // no bid when broken ad markup
+			}
+
+			var bidType openrtb_ext.BidType
+			switch markupType {
```

Comment:
Consider extracting the mapping of markupType to bidType to a method.

Reply:
Good advise, thank you. Made the change.